layers/generate_layers.py

```diff
@@ -71,6 +73,18 @@ def main():
         made_any = True
         makeCensusOutputAreas(args.census_output_areas)
 
+    if args.railway_stations:
```

Comment:
Running `./generate_layers.py -i big_england/england-latest.osm.pbf --railway_stations` gives an error: `TypeError: generateLayerBasedOnTwoPartTag() missing 1 required positional argument: 'outputAsPmtiles'`

I think the new helper functions are making things very confusing. For each of these layers, we should be able to read through one function linearly and understand the steps happening. Helper functions shouldn't be highly nested; our call stack should stay shallow. Some places I'm getting confused:

- What're all the arguments to `generateLayerBasedOnTwoPartTag`? Why are we passing 3 different strings just to construct a tags filter -- why not just directly do "n/railway=station" or "nwr/leisure=pitch,sports_centre" and comment to https://docs.osmcode.org/osmium/latest/osmium-tags-filter.html for reference what these filters mean?
- Why do both `generatePmTilesFromOSMFile` and `generateGeojsonFromOSMFile` quietly call `remove_extra_properties`? Cleaning up properties is orthogonal to converting file formats
- Where can we insert numeric property IDs for railway stations? We still need to do that, otherwise hovering/tooltips isn't going to work.

How about we split out a function for railway stations and list the steps linearly?

```
def makeRailwayStations(osm_input):
    if not osm_input:
        raise Exception("You must specify --osm_input")

    tmp = f"tmp_railway_stations"
    ensureEmptyTempDirectoryExists(tmp)

    run(
        [
            "osmium",
            "tags-filter",
            osm_input,
            "n/railway=station",
            "-o",
            f"{tmp}/extract.osm.pbf",
        ]
    )

    # Transform osm.pbf to GeoJSON, only keeping points.
    # Or we could have a pbfToGeojson(pbf_path, gj_path, geometry_type) helper if repeating the command isn't nice
    run(
        [
            "osmium",
            "export",
            f"{tmp}/extract.osm.pbf",
            "--geometry-type=point",
            "-o",
            f"{tmp}/extract.geojson",
        ]
    )

    # Clean up the file
    print(f"Cleaning up {tmp}/extract.geojson")
    gj = {}
    with open(f"{tmp}/extract.geojson") as f:
        gj = json.load(f)

        counter = 1
        for feature in gj["features"]:
            # Only keep one property
            props = {}
            name = feature["properties"].get("name")
            if name:
                props["name"] = name
            feature["properties"] = props

            feature["geometry"]["coordinates"] = trim_precision(
                feature["geometry"]["coordinates"]
            )

            # The frontend needs IDs for hovering
            feature["id"] = counter
            counter += 1
    with open("output/railway_stations.geojson", "w") as f:
        f.write(json.dumps(gj))
```